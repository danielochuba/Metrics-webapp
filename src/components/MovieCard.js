import React from 'react';
import { Col, Badge } from 'react-bootstrap';
import { PiArrowCircleRightLight } from 'react-icons/pi';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import '../css/movieCard.css';

function MovieCard() {
  const dataItems = useSelector((store) => store.movies.movies);
  const mappedMovies = dataItems.map((movie) => (
    <Col className="movieCard" key={movie.id} xs={6} sm={6} md={4} lg={3}>
      <Link alt="arrow" className="navigator absolute z-10 " to={`/details/${movie.id}`}><PiArrowCircleRightLight alt="arrow" className="absolute text-3xl text-gray-200" /></Link>
      <img className="cardImage" src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`} alt={movie.title} />
      <Badge className="badge" bg="secondary">{ movie.title }</Badge>
    </Col>
  ));

  return (
    <>
      {mappedMovies}
    </>
  );
}

export default MovieCard;
