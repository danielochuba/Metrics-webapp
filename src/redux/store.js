import { configureStore } from '@reduxjs/toolkit';
import countrySlice from './country/countrySlice';

const store = configureStore({
  reducer: {
    movies: countrySlice,
  },
});

export default store;
