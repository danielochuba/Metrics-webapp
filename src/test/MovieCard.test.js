import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import MovieCard from '../components/MovieCard';
import store from '../redux/store';

describe('MovieCards component', () => {
  test('renders without errors', () => {
    render(<Provider store={store}><MovieCard /></Provider>);
  });
});

describe('MovieCard component', () => {
  test('matches the snapshot', () => {
    const { container } = render(<Provider store={store}><MovieCard /></Provider>);
    expect(container).toMatchSnapshot();
  });
});

describe('MovieCard component', () => {
  test('matches the snapshot', () => {
    const movies = [
      { id: 1, title: 'Movie 1', poster_path: '/poster1.jpg' },
      { id: 2, title: 'Movie 2', poster_path: '/poster2.jpg' },
    ];

    const { container } = render(<Provider store={store}><MovieCard movies={movies} /></Provider>);
    expect(container).toMatchSnapshot();
  });
});
