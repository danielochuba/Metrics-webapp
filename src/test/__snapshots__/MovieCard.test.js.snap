// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component matches the snapshot 1`] = `<div />`;

exports[`MovieCard component matches the snapshot 2`] = `<div />`;
